### PR TITLE
Feature/gat 3129 Validation of HDR schema for MK1 simulation server for FMA

### PR DESCRIPTION
**Pull request by shazzzm**
Updated the /api/datasets call to validate the openapi schema. The datasets now come from flat files rather than a mongodb instance.

### CODE REVIEW

File: src/api/v1/services/dataset.service.ts [hunk 1]
```diff
@@ -1,53 +1,91 @@
-import { Datasets } from '../models/dataset.model';
-
 import { dataset } from '../types/dataset.interface';
+import fs from 'fs';
 
 export default class DatasetService {
     public async getDatasets(q: string, offset: number, limit: number): Promise<dataset[]> {
-        const pipeline: Array<any> = [
+        // The files directory contains a set of example JSON files
+        const files = fs.readdirSync("./files");
```
Comment (dannitahdruk) [inline src/api/v1/services/dataset.service.ts:7]: I don't know if you need to use a predefined file vs fetching from the db

Comment (loki-sinclair-hdruk) [reply src/api/v1/services/dataset.service.ts:7]: I asked Tristan to make static files, so we don't have to rely on potentially dodgy data from mk1, @dannitahdruk .

File: src/api/v1/services/dataset.service.ts [hunk 2]
```diff
@@ -1,53 +1,91 @@
...
+                modified : parsedJSON.modified,
+                name : parsedJSON.summary.title,
+                description : parsedJSON.summary.abstract,
+                type : "dataset",
+                source : parsedJSON.summary.publisher.name,
+                self : "http://example-url.com/api/datasets/" + parsedJSON.identifier
+            }
+
+            // If the search query isn't blank
+            if (q != '')
```
Comment (loki-sinclair-hdruk) [inline src/api/v1/services/dataset.service.ts:28]: Sorry, I insist upon strict type checking :) `!==`

File: src/api/v1/services/dataset.service.ts [hunk 3]
```diff
@@ -1,53 +1,91 @@
...
+            {
+                // If the query can be found in the description
+                if (parsedJSON.summary.abstract.search(q) != -1)
```
Comment (loki-sinclair-hdruk) [inline src/api/v1/services/dataset.service.ts:31]: Same here

File: src/api/v1/services/google/logging.service.ts [hunk 4]
```diff
@@ -5,15 +5,29 @@ import Locals from '../../config/locals';
...
             logName: Locals.config().googleLoggingName,
         });
     }
 
     sendDataInLogging(data: any, severity: any) {
-        this._logger.setData(data);
-        this._logger.setSeverity(severity);
-        this._logger.writeLog();
+
+        if (Locals.config().loggingType == "local")
```
Comment (loki-sinclair-hdruk) [inline src/api/v1/services/google/logging.service.ts:19]: And here :)

File: src/api/v1/services/google/logging.service.ts [hunk 5]
```diff
@@ -5,15 +5,29 @@ import Locals from '../../config/locals';
 export default class LoggingService {
     _logger;
     constructor() {
+        if (Locals.config().loggingType != 'gcp') {
```
[the remark this answers is not in the document]
Comment (dannitahdruk) [reply src/api/v1/services/google/logging.service.ts:8]: can you add one env file like example?

**Review by dannitahdruk — APPROVED** [review]

**Review by loki-sinclair-hdruk — APPROVED** [review]

**Pull request merged**